avatars/avatars.js

```diff
@@ -1820,33 +2342,61 @@ class Avatar {
     
     
 
-    _updateHmdPosition();
-    _applyAnimation(this, now, moveFactors);
+    if (this.ragdoll) {
+      if (this.ragdoll !== this.prevRagdoll) {
+        // console.log('switch to ragdoll');
+        if (!this.ragdollMesh) {
+          this.recordCurrentPose();
```

Comment:
Maybe this isn't the best way to initialize the ragdoll.

It seems we need the save/restore only for the purpose of creating the ragdoll mesh? If that's true, why can't we make the t-pose an argument to the ragdoll creation? It doesn't seem like we need to mess with the base model pose at all, so we don't need to save/restore the bones.

Reply:
Instead of "wrap to avatar" it would be "wrap to pose".

If the code was cleaner, then avatar pose could be a `class` and this would be straightforward passing an object.

Reply:
>It seems we need the save/restore only for the purpose of creating the ragdoll mesh? If that's true, why can't we make the t-pose an argument to the ragdoll creation? It doesn't seem like we need to mess with the base model pose at all, so we don't need to save/restore the bones.

Oh, yes, confirmed again that it's no need to reset to TPose to add joint.

The combination of quaternion of `physicsManager.addJoint` and` setJointMotion/setJointSwingLimit` is the real joint setting.

The initial transform has no effect on the joint setup, but does affect the initial velocity as it tries to get back within the proper joint setup limits.
But our avatars bones should always in the limits, so no need to reset to TPose.
Especially I `resetToRecordedPose()` afterwards, so these are really redundant operations, should no difference if delete them.

Reply:
Deleted `recordCurrentPose()`, `resetToRecordedPose()` and `resetToTPose()`. https://github.com/webaverse/app/pull/2731/commits/eaa791fd36428ddc818a8c6de90635a8562c0939

> Instead of "wrap to avatar" it would be "wrap to pose".
> 
> If the code was cleaner, then avatar pose could be a `class Pose` or something and this would be straightforward passing an object.

Hello @avaer, I'm not very sure what `class Pose` and "wrap to pose" do.
Is it still needed after the deletion of `recordCurrentPose()`, `resetToRecordedPose()` and `resetToTPose()`?